pkg/trace/schema/consensus.go

```diff
@@ -166,3 +169,78 @@ func WriteBlockSummary(client trace.Tracer, block *types.Block, size int) {
 		LastCommitRound:          block.LastCommit.Round,
 	})
 }
+
+const (
+	ConsensusStateTable = "consensus_state"
+)
+
+type ConsensusStateUpdateType string
+
+const (
+	ConsensusNewValidBlock      ConsensusStateUpdateType = "new_valid_block"
+	ConsensusNewRoundStep       ConsensusStateUpdateType = "new_round_step"
+	ConsensusVoteSetBits        ConsensusStateUpdateType = "vote_set_bits"
+	ConsensusVoteSet23Prevote   ConsensusStateUpdateType = "vote_set_23_prevote"
+	ConsensusVoteSet23Precommit ConsensusStateUpdateType = "vote_set_23_precommit"
+	ConsensusHasVote            ConsensusStateUpdateType = "has_vote"
+	ConsensusPOL                ConsensusStateUpdateType = "pol"
+)
+
+type ConsensusState struct {
+	Height       int64        `json:"height"`
+	Round        int32        `json:"round"`
+	UpdateType   string       `json:"update_type"`
+	Peer         string       `json:"peer"`
+	TransferType TransferType `json:"transfer_type"`
+}
+
+func (c ConsensusState) Table() string {
+	return ConsensusStateTable
+}
+
+func WriteConsensusState(
+	client trace.Tracer,
+	height int64,
+	round int32,
+	step uint8,
+	peer string,
+	updateType ConsensusStateUpdateType,
+	transferType TransferType,
+) {
+	client.Write(ConsensusState{
+		Height:     height,
+		Round:      round,
+		Peer:       peer,
+		UpdateType: string(updateType),
+	})
+}
+
+const (
+	ProposalTable = "consensus_proposal"
+)
+
+type Proposal struct {
+	Height       int64        `json:"height"`
+	Round        int32        `json:"round"`
+	Peer         string       `json:"peer"`
```

Comment:
Could you clarify what the term 'Peer' refers to for a Proposal message? Does it represent the connection from which a proposal is received, the proposer themselves, or does its meaning vary depending on the type of transfer? Some additional comments for clarification would be helpful.